src/Datadog.Trace.ClrProfiler.Managed/Integrations/ElasticsearchNetCommon.cs

```diff
@@ -27,6 +27,13 @@ public static Scope CreateScope(Tracer tracer, IntegrationInfo integrationId, ob
                 return null;
             }
 
+            var serviceName = $"{tracer.DefaultServiceName}-{ServiceName}";
```

Comment:
I'm worried about every path having this extra allocation even though it'll only be used by a handful of customers.
Can't we instead have a `IsServiceEnabled(string defaultServiceName, string serviceName)` (or add the method on `Tracer` so you can skip the first argument), and do the concatenation only if there actually are excluded services?

Reply:
I don't think this is an additional allocation. We were always concatenating these strings in line 42 below (unless the entire integration was disabled). Zach moved it up because so we could bail out earlier if the service is disabled.

But now that you mention it, can we cache the final service name instead of concatenating every time? Since `ServiceName` is constant, I think we only need to invalidate the cached value if `tracer.DefaultServiceName` changes (which itself can only change if the user replaces `Tracer.Instance`). Probably something for a separate PR.

Reply:
You're right, my bad!